[CS2113-F13-3] NUSplanner

kyrixn opened the pull request:
NUSplanner is a desktop app that allows for an easy and straightforward way for NUS students to manage their schedule ranging from person, school or external related activities. This application makes use of a desktop Command Line Interface (CLI), enabling a quick and sleek method of getting your schedule in check.

quanganh2810 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+
+#### How the feature is implemented:
+
+![Storage Class Diagram](UML/Images/StorageSequenceDiagram.png)
```
The diagram is good but it looks quite hectic. Consider breaking it up using reference frames with proper annotation for easier comprehension

quanganh2810 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+
+The class diagram below illustrates the structure of the storage package
+
+![Storage Class Diagram](UML/Images/StorageClass.png)
```
Simple and easy to read !

quanganh2810 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+
+The class diagram below illustrates the structure of the EventList component.
+
+<img src="UML\Images\EventListUML.png" />
```
Perhaps you should consider noting down the methods within the UI and Parser classes

quanganh2810 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+
+And below is a sequential diagram showing a event being added, revised, checked for information and finally deleted.
+
+<img src="UML\Images\EventListSD.png" style="zoom:80%;" />
```
Good job nice and easy to read !

alexgoexercise replied:
![image](https://user-images.githubusercontent.com/88603534/228767473-ca7f62e1-0940-4eba-8545-3cedebaa477c.png)
there is a red dot in the middle of the diagram, take note!

Masahiro21 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+<img src="UML\Images\EventListUML.png" />
+
+And below is a sequential diagram showing a event being added, revised, checked for information and finally deleted.
+
```
![image](https://user-images.githubusercontent.com/88227374/228456608-6b7b479f-e35f-4d26-b1b8-fc4b90467c78.png)
Not a bug but the arrow is out of place compared to the rest of diagram, to improve on the quality of the diagram you can fix it.

Masahiro21 replied:
![image](https://user-images.githubusercontent.com/88227374/228459677-353f760b-8207-45d2-9db7-7d1d1c0f0c33.png)
![image](https://user-images.githubusercontent.com/88227374/228459778-e11bdfea-d9d5-4d69-9660-bd10624aa178.png)
![image](https://user-images.githubusercontent.com/88227374/228459833-73db5cab-a005-43ec-b7c7-e79256615665.png)

Just some spelling errors in the sequential diagram that can be fixed for better-quality work

Masahiro21 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+![Storage Class Diagram](UML/Images/StorageClass.png)
+
+#### How the feature is implemented:
+
```
![image](https://user-images.githubusercontent.com/88227374/228458501-0a56c7cf-edf3-4e25-9d44-8199573a358b.png)
<<create file>> if file does not exist should be encased in the alt box since it only runs under the condition that the file does not exist

Masahiro21 replied:
![image](https://user-images.githubusercontent.com/88227374/228458927-e2a8f584-338c-46aa-ab07-f30af1b9a6c4.png)

For self-call methods should add another activation bar to prevent confusion

Masahiro21 reviewed (commented):
Overall the diagrams are very detailed and the explanations were easy to understand.

Reynold-SL commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,126 @@
+
+##### Save Events
+
+![Save To File Sequence Diagram](UML/Images/SaveToFile.png)
```
Perhaps the activation bar can be a single one rather than split as the object is created once and called again later?
![image](https://user-images.githubusercontent.com/88631870/228730768-34f1ddf1-6741-473c-b90e-75d8bccd3ac5.png)

Reynold-SL commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,126 @@
+
+##### Load Events
+
+![Load Events Sequence Diagram](UML/Images/loadEvents.png)
```
Perhaps the explanation can be more in depth to allow for a better understanding of the function?

Reynold-SL commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,126 @@
+
+The class diagram below illustrates the structure of the EventList component.
+
+<img src="UML\Images\EventListUML.png" />
```
Perhaps the activation bar can be a single one as the object is created once and called again
![image](https://user-images.githubusercontent.com/88631870/228731838-4a02b932-ef15-4e75-a593-511109be0d81.png)

haoyangw commented on docs/UML/addEvent.puml:
```diff
@@ -0,0 +1,39 @@
+@startuml
+Actor -> Parser : parseAddCommand()
```
I think you should add a `activate Parser` line since the activation bar is missing from your diagram?

haoyangw commented on docs/UML/LoadModules.puml:
```diff
@@ -0,0 +1,36 @@
+-> ":Storage" : loadModules()
+activate ":Storage"
+Alt modulesLoaded
+    ":Storage" --> ":Storage" : :HashMap<String, NusModule>
```
I think this should be a solid arrow since it's a method invocation?

Reynold-SL commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,126 @@
+
+
+The UML diagram below illustrates the flow of how the application adds modules:
+![Add Event Class Diagram](UML/Images/addModules.png)
```
Perhaps you can include an activation bar as the object is created and for better visual understanding?
![image](https://user-images.githubusercontent.com/88631870/228732546-0e9038e8-39a5-4b09-846b-ebb83160a9f1.png)

haoyangw commented on docs/UML/SaveToFile.puml:
```diff
@@ -0,0 +1,37 @@
+    end
+    ":EventListStorage" -> ":FileWriter" : taskWriter.write(gsonData)
+    activate ":FileWriter"
+    ":FileWriter" --> ":File" : String
```
I think this should be a solid arrow since it's a method invocation?

haoyangw commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,126 @@
+Refer to the user guide [here](https://github.com/AY2223S2-CS2113-F13-3/tp/blob/master/docs/UserGuide.md).
+
 
 ## Design & implementation
```
Very clear explanation of the features! Would be helpful to have an explanation of _why_ you implemented your feature logic this way so other developers can understand your considerations.

irving11119 replied:
Perhaps you could include more elaboration such has system architecture and dependencies as well as relevant diagrams for clarity! However, it is already very clear!

haoyangw reviewed (commented):
Very concise and clear explanation of your features! Just some minor syntax issues with the UML diagrams. Great work! :)

irving11119 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,126 @@
+When the application starts up, the storage loadEvents() function will be called to load contents in the save file. 
+
+##### Load Modules
+![Load Modules Sequence Diagram](UML/Images/LoadModules.png)
```
Clear and easy to read UML Diagram!

irving11119 commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,126 @@
+This makes life easier to developers in the future if they wish to add new features that requires users to use new commands.
+
+### Storage Component
+API: `Storage.java`
```
Well formatted and easy to read!

irving11119 commented on docs/UML/addEvent.puml:
```diff
@@ -0,0 +1,39 @@
+Parser -> Ui : addSuccessMsg()
+Ui --> Parser
+Parser --> Actor
+@enduml
```
Perhaps you include the activation bar using `activate`

irving11119 replied:
Additionally, remember to include the cross when you are no longer using the object/ it is deleted.

irving11119 commented on docs/UML/addModules.puml:
```diff
@@ -0,0 +1,21 @@
+    end
+end
+Parser --> Actor
+@enduml
```
Similarly as above, you can include an activation bar by using `activate`

irving11119 commented on docs/UML/LoadEvents.puml:
```diff
@@ -0,0 +1,27 @@
+activate ":Storage"
+":Storage" -> ":EventListStorage" : loadEvents()
+activate ":EventListStorage"
+":EventListStorage" ->":EventListAdapter" : gson.fromJson(fileReader, ArrayList.class)
```
Is this correct? I was under the assumption that when your arrow is drawn to a class, you are calling a method in that class. Is `fromJson()` a method in `:EventListAdapter`?

irving11119 reviewed (commented):
Overall, the developer guide is easy to read and quite clear. However, some of the UML Diagrams can be further improved by following the conventions (e.g having an activation bar)

hrithie committed:
Fixed checkstyle

matthew-liu-zhenjie committed:
Merge pull request #94 from hrithie/hrithieMenon-v2.0

Implemented logger & added logging

alexgoexercise commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+Refer to the user guide [here](https://github.com/AY2223S2-CS2113-F13-3/tp/blob/master/docs/UserGuide.md).
+
 
 ## Design & implementation
```
I like how all the diagrams are in details and that's a whole lotta effort to do it! However it would be greater if you can have an overall architecture!

kyrixn committed:
tmp

kyrixn committed:
handle confliction

kyrixn committed:
Merge branch 'tt'

alexgoexercise commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+
+The Parser component parses the command of the user input and breaks the user input into different parts based on the flags.
+This component also ensures to validate that user input is correct.
+
```
![image](https://user-images.githubusercontent.com/88603534/228765320-97982f7b-e6f0-464f-8d54-5473197b0261.png)
This sequence diagram is missing activation bars, it would be better if you include them

alexgoexercise commented on docs/DeveloperGuide.md:
```diff
@@ -1,15 +1,93 @@
+
+When the application starts up, the storage loadEvents() function will be called to load contents in the save file. 
+Similarly, the state of the user's event list is saved when the user exits the application by calling saveToFile().
+
```
![image](https://user-images.githubusercontent.com/88603534/228766317-76a5410c-321a-49c7-8ab0-6ac547335bc6.png)
I would recommend using a constructor notation like what we have learnt in this module instead of "<<create>>" notation for creating a constructor.

kyrixn committed:
add format check for -r

matthew-liu-zhenjie committed:
Update UserGuide

kyrixn committed:
start revising UserUtility

matthew-liu-zhenjie committed:
Fix typo in UserGuide

matthew-liu-zhenjie committed:
Add Architecture class diagram

matthew-liu-zhenjie committed:
Update Developer Guide

Add User Stories, Add Architecture documentation and dependencies.

matthew-liu-zhenjie committed:
Add NFR to Developer Guide

hrithie committed:
Merge pull request #95 from matthew-liu-zhenjie/Documentation

Documentation

hrithie committed:
Added to DG

matthew-liu-zhenjie committed:
Fix bug where .jar does not load nusmods data

matthew-liu-zhenjie committed:
Remove debugging println

matthew-liu-zhenjie committed:
Add JUnit test for StorageManager

matthew-liu-zhenjie committed:
Update EventListAdapter

kyrixn committed:
Merge pull request #96 from hrithie/hrithieMenon-v2.0

Added to DG

kyrixn committed:
Merge pull request #97 from matthew-liu-zhenjie/BugFixStorage

Bug fix storage

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into StorageUpgrade

kyrixn committed:
writing recurring evvent in tt, haven't finished

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into new-pr

matthew-liu-zhenjie committed:
Improve defensiveness of Storage

matthew-liu-zhenjie committed:
Fix Checkstyle error

matthew-liu-zhenjie committed:
Update UserGuide

Edit command summary

kyrixn committed:
finished adding recurring event to tt

hrithie committed:
Fixes to Ui (Incomplete)

kyrixn committed:
Merge pull request #99 from matthew-liu-zhenjie/UserGuide

Update UserGuide

kyrixn committed:
Merge pull request #98 from matthew-liu-zhenjie/StorageUpgrade

Storage upgrade

matthew-liu-zhenjie committed:
Merge pull request #100 from kyrixn/tt-for-recurring

show recurring event in timetable

hrithie committed:
Checkstyle fix

hrithie committed:
Checkstyle fix

kyrixn committed:
Merge pull request #101 from hrithie/hrithieMenon-v2.0

Fixes to Ui (Incomplete)

kyrixn committed:
fix confliction

kyrixn committed:
Merge pull request #102 from kyrixn/tt-for-recurring

fix confliction

matthew-liu-zhenjie committed:
Merge pull request #103 from AY2223S2-CS2113-F13-3/show-recurring-in-tt

fixed confliction

hrithie committed:
Fix week for timetable

matthew-liu-zhenjie committed:
Merge pull request #104 from hrithie/hrithieMenon-v2.0

Fix week for timetable

hrithie committed:
Timetable fix

kyrixn committed:
tmp

kyrixn committed:
added log to gitignore

kyrixn committed:
Merge branch 'tt-for-recurring'

kyrixn committed:
fix PE-D issue 127: end time ommited, and extracted some function to improve readability

kyrixn committed:
Merge branch 'tt-for-recurring'

kyrixn committed:
Merge pull request #139 from hrithie/hrithieMenon-v2.0

Ui timetable fix

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into tt-for-recurring

kyrixn committed:
Merge branch 'tt-for-recurring'

matthew-liu-zhenjie committed:
Merge pull request #140 from kyrixn/master

fixed issue 127

hrithie committed:
Fix for issue #110 Able to delete all even though list is empty

hrithie committed:
Merge branch 'master' into hrithieMenon-v2.0

hrithie committed:
Update Ui.java

matthew-liu-zhenjie committed:
Merge pull request #145 from hrithie/hrithieMenon-v2.0

Fix for issue #110 Able to delete all even though list is empty

matthew-liu-zhenjie committed:
Fix UML diagrams

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

kyrixn committed:
fixed error that cannot add module

hrithie committed:
Added JavaDoc and Fix for #114 input not trimmed

hrithie committed:
Fixed 2 tasks from TA code review

hrithie committed:
Fix checkstyle

hrithie committed:
Fixed 1 task from TA code review

kyrixn committed:
Merge pull request #146 from hrithie/hrithieMenon-v2.0

Fixes and JavaDoc

kyrixn committed:
Merge branch 'master' into master

kyrixn committed:
Update Parser.java

kyrixn committed:
resolved conflict locally

kyrixn committed:
resolved conflict locally

kyrixn committed:
Update Parser.java

hrithie committed:
Fixed issue #137

hrithie committed:
Merge pull request #148 from kyrixn/master

fixed error that app cannot detect adding modules

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

matthew-liu-zhenjie committed:
Fix parseDeleteCommand

Fix bug where "delete -a" printed an unexpected message. Bug where "delete -s" for negative integers resulted in unexpected error message.

matthew-liu-zhenjie committed:
Fix bug in parseAddCommand

Fix bug where moduleCode is not converted to uppercase before being searched leading to module not being found in the hashmap.

matthew-liu-zhenjie committed:
Add PPP

matthew-liu-zhenjie committed:
Merge pull request #149 from hrithie/hrithieMenon-v2.0

Fixing issues

hrithie committed:
Merge pull request #150 from matthew-liu-zhenjie/master

Fix UG bugs

hrithie committed:
Merge pull request #151 from matthew-liu-zhenjie/Issue118

Fix bug in parseAddCommand

hrithie committed:
Merge pull request #152 from matthew-liu-zhenjie/Issue124

Fix parseDeleteCommand

hrithie committed:
Merge pull request #153 from matthew-liu-zhenjie/Matthew_PPP

Add PPP

kyrixn committed:
finished code for confliction check

kyrixn committed:
finished conliction check

matthew-liu-zhenjie committed:
Merge pull request #154 from kyrixn/check-confliction

finished confliction check and tested on some cases

kyrixn committed:
try debugging

kyrixn committed:
finished checking confliction, tested on all cases

kyrixn committed:
Update EventList.java

matthew-liu-zhenjie committed:
Merge pull request #155 from kyrixn/check-confliction

finished Checking confliction

matthew-liu-zhenjie committed:
Fix Storage JUnit tests

matthew-liu-zhenjie committed:
Fix JUnit test for storage

matthew-liu-zhenjie committed:
Fix Checkstyle Errors

kyrixn committed:
Merge pull request #156 from matthew-liu-zhenjie/FixTest

Fix test

kyrixn committed:
fixed bugs in adding modules that conflicts with other events

kyrixn committed:
resolved confliction locally

kyrixn committed:
fixed bugs in checking confliction and updated UserGuide

kyrixn committed:
Update Schedule.java

Zemdalk committed:
use static final variables to replace string literals

Zemdalk committed:
refactor: extract a general Schedule constructor, fix some style issues

matthew-liu-zhenjie committed:
Fix JsonEventListStorage
Fixed potential bug where user crashes application if they clear save.json

matthew-liu-zhenjie committed:
Refactor Storage based on #144

kyrixn committed:
Merge pull request #157 from kyrixn/master

fixed bugs and updated UG

matthew-liu-zhenjie committed:
Add support for special term modules

Implemented parsing for week object from nusmods.json

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into StorageImprovement

matthew-liu-zhenjie committed:
Update UG

matthew-liu-zhenjie committed:
Fix Checksytle Error

Remove unused import

matthew-liu-zhenjie committed:
Update ModuleAdapter

prevent case where weeks array is already present but is added on again

hrithie committed:
Merge pull request #158 from matthew-liu-zhenjie/StorageImprovement

Storage improvement

Zemdalk committed:
refactor: extract a general Schedule constructor, fix some style issues

matthew-liu-zhenjie committed:
Fix bug in EventListAdapter
fix bug where event with no start time can be added after modifications to json file.

Zemdalk committed:
fix some style issues

hrithie committed:
Merge pull request #159 from matthew-liu-zhenjie/master

Fix bug in EventListAdapter

Zemdalk committed:
fix merge conflicts

hrithie committed:
Merge pull request #161 from Zemdalk/master

Refactor contructors in Schedule, write some java doc & add more details in exception message

matthew-liu-zhenjie committed:
Update DG

matthew-liu-zhenjie committed:
Update DG

Add contents page, fleshed out manual testing instructions.

matthew-liu-zhenjie committed:
Update DG
Elaborated on design consdierations

matthew-liu-zhenjie committed:
Update DG

matthew-liu-zhenjie committed:
Update EventListAdapter

Catch potential bug where event endTime is before event startTime.

hrithie committed:
First draft

matthew-liu-zhenjie committed:
Merge pull request #164 from hrithie/documentation

First draft

kyrixn committed:
Merge pull request #162 from matthew-liu-zhenjie/DevGuide

kyrixn committed:
Merge pull request #163 from matthew-liu-zhenjie/StorageAgain

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

Fix broken contents page links

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

Fix broken introduction link

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

Formatting

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

Formatting

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

Test page break

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

test format

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

format addevent image

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

format addevent

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

matthew-liu-zhenjie committed:
Update DeveloperGuide.md

hrithie committed:
Fix: checks if module is in correct sem

kyrixn committed:
handle exceptions for illegal reccurring time

kyrixn committed:
Merge branch 'master' of https://github.com/kyrixn/tp

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

kyrixn committed:
update UG

matthew-liu-zhenjie committed:
Resize UML diagrams, update DG pagebreak

matthew-liu-zhenjie committed:
Update DG

improve formatting

matthew-liu-zhenjie committed:
Update DG

matthew-liu-zhenjie committed:
Update DG

matthew-liu-zhenjie committed:
Merge pull request #165 from hrithie/documentation

Fix: checks if module is in correct sem

matthew-liu-zhenjie committed:
Update DG, UG, Storage UML

hrithie committed:
Merge pull request #166 from matthew-liu-zhenjie/DevGuide

Dev guide

matthew-liu-zhenjie committed:
Fix bug in SaveToFile UML

matthew-liu-zhenjie committed:
Fix bug in SaveToFile UML diagram

Destroy FileWriter at end of process

matthew-liu-zhenjie committed:
Fix UML

Reverse previous commit

kyrixn committed:
minor changes and draft of PPP

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

kyrixn committed:
changed DG

kyrixn committed:
minor changes, tidy up the code

matthew-liu-zhenjie committed:
Update UG and Storage

Update storage lessonType key to match what is displayed on NUSMODS

matthew-liu-zhenjie committed:
Update UG

Reflect changes in code in UG

kyrixn committed:
Merge pull request #170 from matthew-liu-zhenjie/master

Update UG and Storage

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into DevGuide

matthew-liu-zhenjie committed:
Update Matthew.md

Changed back reposense link

matthew-liu-zhenjie committed:
Merge pull request #169 from kyrixn/master

minor changes and uplaoded PPP draft

matthew-liu-zhenjie committed:
Merge branch 'exceptions' into MergeExceptions

hrithie committed:
Merge pull request #171 from matthew-liu-zhenjie/MergeExceptions

Merge exceptions from upstream to master.

matthew-liu-zhenjie committed:
Merge branch 'master' into DevGuide

hrithie committed:
Fix infinite loop for getSemester()

hrithie committed:
Update EXPECTED.TXT

kyrixn committed:
fixed confliction checking bugs and finished ppp

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

kyrixn committed:
editedd DG

matthew-liu-zhenjie committed:
Merge pull request #173 from kyrixn/master

fixed bug and update DG

hrithie committed:
Fix

matthew-liu-zhenjie committed:
Update Ui.java

hrithie committed:
Add help command

hrithie committed:
Update Ui.java

hrithie committed:
Fix checkstyle

matthew-liu-zhenjie committed:
Fix bug where conflicting events can be made in JSON save file.

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into FixStorageConflictBug

matthew-liu-zhenjie committed:
Merge pull request #174 from hrithie/hrithieMenon-v2.1

Add help command

matthew-liu-zhenjie committed:
Merge pull request #172 from hrithie/master

Fix infinite loop for getSemester()

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into FixStorageConflictBug

hrithie committed:
Update PPP

matthew-liu-zhenjie committed:
Fix confliction in EventList

matthew-liu-zhenjie committed:
Fix bug where error message is errornously printed out.

hrithie committed:
Resolved issues in UserGuide.md

hrithie committed:
Merge pull request #175 from matthew-liu-zhenjie/FixStorageConflictBug

Fix storage conflict bug

kyrixn committed:
minor changes

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

hrithie committed:
Updated new output

hrithie committed:
Resolve issue

matthew-liu-zhenjie committed:
Merge pull request #176 from hrithie/hrithieMenon-v2.1

Update UserGuide and PPP

matthew-liu-zhenjie committed:
Merge branch 'master' into DevGuide

matthew-liu-zhenjie committed:
Update UG

Format for PDF

matthew-liu-zhenjie committed:
Fix UG

Fix format for 1.0

kyrixn committed:
handled illegal week number

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

matthew-liu-zhenjie committed:
Format UG

matthew-liu-zhenjie committed:
Merge pull request #177 from kyrixn/master

handled illegal week number input

matthew-liu-zhenjie committed:
Update UG

Add page breaks

kyrixn committed:
renamed constant var

matthew-liu-zhenjie committed:
Update UG

matthew-liu-zhenjie committed:
Fix UG bugs

matthew-liu-zhenjie committed:
Fix DG bugs

matthew-liu-zhenjie committed:
Fix DG bugs

Update UG link and remove page break.

matthew-liu-zhenjie committed:
Reformat Storage component in DG

matthew-liu-zhenjie committed:
Update UG

matthew-liu-zhenjie committed:
Update UG

matthew-liu-zhenjie committed:
Fix DG bug

Change documentation link to latest in Acknowledgements for GSON.

matthew-liu-zhenjie committed:
Fix DG bug

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into DevGuide

kyrixn committed:
Merge pull request #178 from matthew-liu-zhenjie/DevGuide

User & Dev guide

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

kyrixn committed:
add time check in reviseTimeInfo, correccted accepted time format for add and edit commmand in UG

matthew-liu-zhenjie committed:
Merge pull request #179 from kyrixn/master

changed UG and added time check in reviseTimeInfo()

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp into DevGuide

matthew-liu-zhenjie committed:
Update UG

matthew-liu-zhenjie committed:
Fix UG

matthew-liu-zhenjie committed:
Fix Storage Class Diagram

remove packages that are not used.

kyrixn committed:
Merge pull request #180 from matthew-liu-zhenjie/DevGuide

UG fixes

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

matthew-liu-zhenjie committed:
Edit Ui help command

Prevent Encoding error on bullet points.

matthew-liu-zhenjie committed:
Update DG

matthew-liu-zhenjie committed:
Add JavaDoc

matthew-liu-zhenjie committed:
Update DG

Specified when nusmods.json data was last accurate.

kyrixn committed:
fix bugs in delete

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

kyrixn committed:
Merge pull request #181 from matthew-liu-zhenjie/master

Fix DG Bug

kyrixn committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

matthew-liu-zhenjie committed:
Merge pull request #182 from kyrixn/master

fix bug

matthew-liu-zhenjie committed:
Update DG

Improved specificity

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

matthew-liu-zhenjie committed:
Update DG

Formatting

matthew-liu-zhenjie committed:
Update DG

Fix Bugs

matthew-liu-zhenjie committed:
Fix DG

Removed typo

matthew-liu-zhenjie committed:
Fix bugs in DG

matthew-liu-zhenjie committed:
Improve DG

matthew-liu-zhenjie committed:
Fix typo in DG, update PPP

matthew-liu-zhenjie committed:
Add PPP

matthew-liu-zhenjie committed:
Update PPP

matthew-liu-zhenjie committed:
Update PPP

added on to team-based tasks

hrithie committed:
DG Fix

hrithie committed:
Merge pull request #183 from matthew-liu-zhenjie/master

Update DG

matthew-liu-zhenjie committed:
Fix Bug in parseDeleteCommand

matthew-liu-zhenjie committed:
Merge pull request #184 from hrithie/hrithieMenon-v2.1

DG Fix

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

hrithie committed:
Merge pull request #185 from matthew-liu-zhenjie/master

Fix Bug in parseDeleteCommand

matthew-liu-zhenjie committed:
Fix bug for empty description or whitespace desription

hrithie committed:
Merge pull request #186 from matthew-liu-zhenjie/master

Fix bug for empty description or whitespace desription

Zemdalk committed:
add ppp

Zemdalk committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

Zemdalk committed:
Merge pull request #188 from Zemdalk/master

add Xingyu's PPP

matthew-liu-zhenjie committed:
add assertions

matthew-liu-zhenjie committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F13-3/tp

hrithie committed:
Merge pull request #189 from matthew-liu-zhenjie/master
